std/conv.d

```diff
@@ -2184,7 +2187,7 @@ Note:
     All character input range conversions using $(LREF to) are forwarded
     to `parse` and do not require lvalues.
 */
-Target parse(Target, Source)(ref Source source)
+auto parse(Target, Source, Flag!"doCount" doCount = No.doCount)(ref Source source)
```

Comment:
can you try 

```d
auto parse(Target, Flag!"doCount" doCount = No.doCount, Source)(ref Source source)
```
instead of 
```d
auto parse(Target, Source, Flag!"doCount" doCount = No.doCount)(ref Source source)
```
so that you only need to do 
```d
T t = parse!(T,Yes.doCount)(str);
```
rather than 
```d
T t = parse!(T, string, Yes.doCount)(str);
```
and update any explicit instantiation unit tests. I want to see if any external actually use explicit instantiation.

Reply:
That was a great suggestion, I addressed it.

Reply:
So, it looks like there are errors. How do I continue from here? Do I go back and redo the order of the parameters, or do I try to modify the code that makes it fail? If the latter, can you give me a push in the right direction? (I never used dub).

Reply:
> can you try

> `auto parse(Target, Flag!"doCount" doCount = No.doCount, Source)(ref Source source)`

@thewilsonator this won't work as this will be a breaking change for code that was previously explicit about it's `Target` and `Source`.

I don't have an example at hand, but I'm sure there is code out there that does `parse!(Target, Source)(..)`

Reply:
> That was a great suggestion, I addressed it.

@adelavais I think you should revert the code to the original order of the template arguments.

Reply:
@edi33416 of course that instead of changing the commit I just re-pushed the same one. Fixed.

Reply:
> but I'm sure there is code out there that does parse!(Target, Source)(..)

I highly doubt it. `Source` is inferred by the first argument, which must always be supplied.

Reply:
@edi33416 the only use I can think of for specifying the source is if for some reason you want to have the source be implicitly cast to or constructed as a different source. Given that the buildkite passed before, it suggests that nobody explicitly instantiates `Source`.

Reply:
@thewilsonator but some tests did fail at the last commit, with the signature
  auto parse(Target, Flag!"doCount" doCount = No.doCount, Source)(ref Source source)
Have a look at https://buildkite.com/dlang/phobos/builds/3945#47569863-a57d-45b5-a0f2-f052ba417631

Reply:
> > but I'm sure there is code out there that does parse!(Target, Source)(..)
> 
> I highly doubt it. `Source` is inferred by the first argument, which must always be supplied.

@thewilsonator I think your "doubt" of being used here isn't a very good reason to introduce breaking changes. I can certainly think of reasons to do this like aliasing `alias parseInt = parse!(int, string);` (or similar) and other generic template code. If not with `parse` I have certainly done this with `to` before. I don't have any code where this is actually done but I think (especially with `parse`) this might be done more often than never.

Reply:
I'm marking this as resolved